dot/parachain/util/util.go

```diff
@@ -140,3 +143,29 @@ func GetBlockAncestors(
 
 	return response.Ancestors, nil
 }
+
+func ExecutorParamsAtRelayParent(rt runtime.Instance, relayParent common.Hash,
+) (*parachaintypes.ExecutorParams, error) {
+	sessionIndex, err := rt.ParachainHostSessionIndexForChild()
+	if err != nil {
+		return nil, fmt.Errorf("getting session index for relay parent %s: %w", relayParent, err)
+	}
+
+	executorParams, err := rt.ParachainHostSessionExecutorParams(sessionIndex)
+	if err != nil {
+		if errors.Is(err, wazero_runtime.ErrExportFunctionNotFound) {
+			// Runtime doesn't yet support the api requested,
+			// should execute anyway with default set of parameters.
+			defaultExecutorParams := parachaintypes.NewExecutorParams()
+			return &defaultExecutorParams, nil
+		}
+		return nil, err
+	}
+
+	if executorParams == nil {
+		// should never happen
+		return nil, fmt.Errorf("executor params for relay parent %s is nil", relayParent)
```

Comment:
should we panic here then? I would say that unexpected behavior can be considered as panics